### PR TITLE
Improved joystick support

### PR DESCRIPTION
**Pull request by macsforme**
This PR implements improved joystick handling per my comments [here on issue #72](https://github.com/BZFlag-Dev/bzflag/issues/72#issuecomment-549397191).

[This video](https://youtu.be/5pwImEa5vf8) explains the changes better than I could by typing it out here, so I will leave it at that.

I tested this on my SteelSeries Nimbus, but it would help if anyone with a different model of joystick or controller could also give it a whirl and make sure the effects of the joystick input settings are as expected.

### CODE REVIEW

**Comment by Ratfink** [comment]
The ramp options in the menu aren't named quite right, since they're really using a polynomial function, not exponential.  I'd suggest being more concise and just calling the three options "Linear," "Quadratic," and "Cubic."

File: src/platform/DXJoystick.cxx
Comment (jwmelto) [inline src/platform/DXJoystick.cxx:264]: lots of superfluous casting here (but at least they are function-style casts and not C-style casts)... the divide by float will do the type promotion, and the type of `x` will narrow as needed.

**Review by jwmelto — APPROVED** [review]
I wonder if we want to persist in 32-bit floating point interfaces/calculations ... all modern hardware (except for GPU) is faster in 64-bit operations. What are we gaining by constraining precision?

File: src/platform/DXJoystick.cxx
Comment (macsforme) [reply src/platform/DXJoystick.cxx]: Fixed... thanks.

**Comment by macsforme** [comment]
This project kind of stalled after I learned that one of the most common controllers out there (the Xbox One controller) has triggers which are treated as axes rather than buttons.  Our code currently has no support for the concept of an axis triggering a button event, so this makes the triggers on the Xbox One controller basically useless for their intended purpose.  Implementing support for an axis to trigger a button would require consideration of actuation point, release point, etc.  Expanding the underlying joystick support code is complicated by the number of platform implementations we have on top of it (EvDev, USB, Windows, SDL 1, SDL 2).  Development time and available hardware for testing are also scarce resources.

Any thoughts on how to move forward would be welcome.  One valid option may be to simply merge as-is, and ignore the lack of axis->button support for the time being, since we will be no worse off than we were before this patch.  **However**, the EvDev/USB/Windows joystick code should be updated to use the new floating-point axis readings versus the old integer readings (I believe Windows is done, but may need testing), if any of that code is even connected to anything anymore.

**Comment by blast007** [comment]
We should remove the non-SDL2 joystick code. That was my plan at least in master once we're done worrying about merging in 2.4 changes. The main reason the Windows joystick code is still in use is because of its Force Feedback support. I do have the start of code that added haptic feedback to our SDL2 joystick code, though.

I was considering an analog-to-digital conversion already to handle menu/UI navigation with analog sticks.

We could also look into use the [GameController API](https://wiki.libsdl.org/CategoryGameController) that SDL2 provides which has mappings for [various controllers](https://raw.githubusercontent.com/gabomdq/SDL_GameControllerDB/master/gamecontrollerdb.txt).

**Comment by macsforme** [comment]
That last commit ([dcc3d87](https://github.com/BZFlag-Dev/bzflag/pull/255/commits/dcc3d87b0a589b88c2d85961405b8ec374581d66)) fixes my last concern with this PR, which was an issue where the real coordinates and effective coordinates did not always align due to incorrect math. [This video](https://youtu.be/Seot2qzjORs) shows that the two are now properly aligned (the red/green lines were a temporary visual aid and are not present in the main PR). Due to the math change, a second set of eyes on that commit would be welcome.

With that, I believe this is ready to merge assuming no other outstanding concerns.

**Pull request merged**
